src/common/ini_handler.c

```diff
@@ -18,32 +18,60 @@ static int ini_handler_func(void *user, const char *section, const char *name, c
 
     ETH_config_t *tmp_config = (ETH_config_t *)user;
 
-    if (ini_string_match(section, "lan_ip4", name, "type")) {
-        if (strncasecmp(value, "DHCP", 4) == 0) {
+    if (ini_string_match(section, "eth::ipv4", name, "type")) {
+        if (strncmp(value, "DHCP", 4) == 0) {
```

Comment:
I still don't agree with enforcing just the uppercase `DHCP` and `STATIC`. An ordinary user will have more trouble making it run. What was wrong with accepting case-insensitive `dhcp` vs. `DHCP`?

Reply:
ok